File: src/main/java/com/chatdemo/ProviderConfig.java
package com.chatdemo;

/**
 * Configuration record for an AI provider.
 * 
 * @param providerType The type of provider (gemini, chatgpt, claude, grok)
 * @param model The model identifier to use
 * @param apiKey The API key for authentication
 */
public record ProviderConfig(
    String providerType,
    String model,
    String apiKey
) {
    /**
     * Get a display name for this configuration.
     * 
     * @return Formatted display name like "ChatGPT (gpt-4o)"
     */
    public String getDisplayName() {
        String name = switch (providerType.toLowerCase()) {
            case "gemini" -> "Gemini";
            case "chatgpt" -> "ChatGPT";
            case "claude" -> "Claude";
            case "grok" -> "Grok";
            default -> providerType;
        };
        return name + " (" + model + ")";
    }
}

File: src/main/scala/com/chatdemo/common/config/ProviderConfig.scala
package com.chatdemo.common.config

/**
 * Configuration for an AI provider.
 *
 * @param providerType The type of provider (gemini, chatgpt, claude, grok)
 * @param model        The model identifier to use
 * @param apiKey       The API key for authentication
 */
case class ProviderConfig(providerType: String, model: String, apiKey: String) {

  /** Formatted display name like "ChatGPT (gpt-4o)". */
  def getDisplayName: String = {
    val name = providerType.toLowerCase match {
      case "gemini"  => "Gemini"
      case "chatgpt" => "ChatGPT"
      case "claude"  => "Claude"
      case "grok"    => "Grok"
      case _         => providerType
    }
    s"$name ($model)"
  }
}
